File: app/src/main/java/org/ole/planet/myplanet/ui/userprofile/AdapterOtherInfo.java
package org.ole.planet.myplanet.ui.userprofile;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.gson.Gson;
import com.google.gson.JsonObject;

import org.ole.planet.myplanet.databinding.RowOtherInfoBinding;
import org.ole.planet.myplanet.utilities.JsonUtils;

import java.util.List;

public class AdapterOtherInfo extends RecyclerView.Adapter<AdapterOtherInfo.ViewHolderOtherInfo> {
    private RowOtherInfoBinding rowOtherInfoBinding;
    private Context context;
    private List<String> list;

    public AdapterOtherInfo(Context context, List<String> list) {
        this.context = context;
        this.list = list;
    }

    @NonNull
    @Override
    public ViewHolderOtherInfo onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        rowOtherInfoBinding = RowOtherInfoBinding.inflate(LayoutInflater.from(context), parent, false);
        return new ViewHolderOtherInfo(rowOtherInfoBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolderOtherInfo holder, int position) {
        if (position < list.size()) {
            String jsonString = list.get(position);
            JsonObject object = new Gson().fromJson(jsonString, JsonObject.class);
            String res = JsonUtils.getString("name", object) + "\n" +
                    JsonUtils.getString("relationship", object) + "\n" +
                    JsonUtils.getString("phone", object) + "\n" +
                    JsonUtils.getString("email", object) + "\n";
            holder.rowOtherInfoBinding.tvDescription.setText(res);
        }
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public static class ViewHolderOtherInfo extends RecyclerView.ViewHolder {
        public RowOtherInfoBinding rowOtherInfoBinding;

        public ViewHolderOtherInfo(RowOtherInfoBinding rowOtherInfoBinding) {
            super(rowOtherInfoBinding.getRoot());
            this.rowOtherInfoBinding = rowOtherInfoBinding;
        }
    }
}

File: app/src/main/java/org/ole/planet/myplanet/ui/userprofile/AdapterOtherInfo.kt
package org.ole.planet.myplanet.ui.userprofile

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.google.gson.JsonObject
import io.realm.RealmList
import org.ole.planet.myplanet.utilities.GsonUtils
import org.ole.planet.myplanet.databinding.RowOtherInfoBinding
import org.ole.planet.myplanet.ui.userprofile.AdapterOtherInfo.ViewHolderOtherInfo
import org.ole.planet.myplanet.utilities.JsonUtils.getString

class AdapterOtherInfo(private val context: Context, private val list: RealmList<String>) :
    RecyclerView.Adapter<ViewHolderOtherInfo>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolderOtherInfo {
        val binding = RowOtherInfoBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        )
        return ViewHolderOtherInfo(binding)
    }

    override fun onBindViewHolder(holder: ViewHolderOtherInfo, position: Int) {
        if (position < list.size) {
            val jsonString = list[position]
            val `object` = GsonUtils.gson.fromJson(jsonString, JsonObject::class.java)
            val res = """
                ${getString("name", `object`)}
                ${getString("relationship", `object`)}
                ${getString("phone", `object`)}
                ${getString("email", `object`)}
                """.trimIndent()
            holder.rowOtherInfoBinding.tvDescription.text = res
        }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    class ViewHolderOtherInfo(var rowOtherInfoBinding: RowOtherInfoBinding) : RecyclerView.ViewHolder(rowOtherInfoBinding.root)
}
